odoo/http.py

```diff
@@ -418,20 +347,173 @@ def serialize_exception(exception):
         'context': getattr(exception, 'context', {}),
     }
 
-def set_safe_image_headers(headers, content):
-    """Return new headers based on `headers` but with `Content-Length` and
-    `Content-Type` set appropriately depending on the given `content` only if it
-    is safe to do, as well as `X-Content-Type-Options: nosniff` so that if the
-    file is of an unsafe type, it is not interpreted as that type if the
-    `Content-type` header was already set to a different mimetype
+
+# =========================================================
+# File Streaming
+# =========================================================
+
+def send_file(filepath_or_fp, mimetype=None, as_attachment=False, filename=None, mtime=None,
+              add_etags=True, cache_timeout=STATIC_CACHE, conditional=True):
+    warnings.warn('odoo.http.send_file is deprecated, please use odoo.http.Stream instead.', DeprecationWarning, stacklevel=2)
+    return _send_file(
+        filepath_or_fp,
+        request.httprequest.environ,
+        mimetype=mimetype,
+        as_attachment=as_attachment,
+        download_name=filename,
+        last_modified=mtime,
+        etag=add_etags,
+        max_age=cache_timeout,
+        conditional=conditional
+    )
+
+
+class Stream:
     """
-    headers = werkzeug.datastructures.Headers(headers)
-    content_type = guess_mimetype(content)
-    if content_type in SAFE_IMAGE_MIMETYPES:
-        headers['Content-Type'] = content_type
-    headers['X-Content-Type-Options'] = 'nosniff'
-    headers['Content-Length'] = len(content)
-    return list(headers)
+    Send the content of a file, an attachment or a binary field via HTTP
+
+    This utility is safe, cache-aware and uses the best available
+    streaming strategy. Works best with the --x-sendfile cli option.
+
+    Create a Stream via one of the constructors: :meth:`~from_path`:,
+    :meth:`~from_attachment`: or :meth:`~from_binary_field`:, generate
+    the corresponding HTTP response object via :meth:`~get_response`:.
+
+    Instantiating a Stream object manually without using one of the
+    dedicated constructors is discouraged.
+    """
+
+    type: str = ''  # 'data' or 'path' or 'url'
+    data = None
+    path = None
+    url = None
+
+    mimetype = None
+    as_attachment = False
+    download_name = None
+    conditional = True
+    etag = True
+    last_modified = None
+    max_age = None
+    size = None
+
+    def __init__(self, **kwargs):
+        self.__dict__.update(kwargs)
+
+    @classmethod
+    def from_path(cls, path, filter_ext=('',)):
+        """ Create a :class:`~Stream`: from an addon resource. """
+        path = file_path(path, filter_ext)
+        check = adler32(path.encode("utf-8")) & 0xFFFFFFFF
```

Comment:
```suggestion
        check = adler32(path.encode())
```
> Changed in version 3.0: The result is always unsigned. To generate the same numeric value when using Python 2 or earlier, use `adler32(data) & 0xffffffff`.

We haven't cared about "Python 2 or earlier" in quite a while now.